webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/AppConfigAuth.scala

```diff
@@ -418,6 +447,55 @@ class RudderXmlUserDetailsContextMapper(authConfigProvider: UserDetailListProvid
   }
 }
 
+/**
+  * A description of some properties of an authentication backend
+  * @param hasAdditionalRoles if the backend can provide additional roles than the ones of the default backend 
+  * @param overridesRoles if the backend is configured to override the roles of the default backend and cannot change them
+  * @param hasModifiablePassword if the backend is configured to allow password modification
+  */
+final case class AuthBackendProviderProperties private (
+    hasAdditionalRoles:    Boolean,
+    overridesRoles:        Boolean,
+    hasModifiablePassword: Boolean
```

Comment:
I think the logic here is not clear (or I don't understand it). 
LDAP does not have updatable password by rudder: they are managed by the LDAP directory, out of Rudder domain of responsibility. 
But if we have a configuration with `provider = ldap, file`, ie first trying LDAP, then trying file, then we want to let the user password for the *file* case be updatable. 
But `file` is not defined as an `AuthBackendProviderProperties`. Why so ?

Reply:
Maybe the `hasModifiablePassword` should be renamed `canProviderBeMigratedToFile` because that is what happens with LDAP...

I didn't really want to introduce this boolean, but its sole purpose is to allow the UI to "force" a password for an external provider (which LDAP is), as in this Elm code : https://github.com/clarktsiory/rudder-plugins/blob/bug_24219/display_user_detail_using_information_from_database/user-management/src/main/elm/sources/View.elm#L144.

Indeed with `provider = ldap, file` we want to let the user password for the file case be updatable, but we don't want it to be updatable with `provider = oidc, file` (unless I was misunderstanding the spec), so a boolean attribute for making distinction between both would be actually needed

Reply:
`file` is always handled very specifically even in the UI, so I didn't want to abstract all the properties that make the distinction between file and other providers, and I guess there are plenty...

Reply:
you can completely have `provider = oidc, file` with updatable passwords *safe* oidc provisioning is set to true (for at least one provider, because I don't think can know from which oidc provider users come... Which is perhaps a problem in fact - for that case, and for the case where a new OIDC provider is added or an old one removed...). 
Ok, appart that new interesting subject: an oidc provider without auto-provisioning is almost exactly like LDAP conceptually: user are declared in `rudder-users.xml`, but the authentication is done by some other entity. 

And LDAP could, one day, with lots of incentives, get an auto-provisioning mode.